src/mpi/comm/commutil.c

```diff
@@ -665,6 +665,9 @@ int MPIR_Comm_create_subcomms(MPIR_Comm * comm)
     MPIR_Assert(num_local > 1 || external_rank >= 0);
     MPIR_Assert(external_rank < 0 || external_procs != NULL);
 
+    comm->num_local = num_local;
+    comm->num_external = num_external;
```

Comment:
`num_local` and `num_external` outside the context of this function are difficult to decipher, IMO. It would be good to include a _what_. E.g. `num_local_procs` and `num_external_nodes`. But does `num_external` also includes the local node?

Reply:
Both are caching the results from `MPIR_Find_{local,external}`, along with `intranode_table` and `internode_table`. The meaning of all of them are tied to the semantics of `MPIR_Find_{local,external}`, which are difficult to decipher other than reading the code. I can group them in `mpir_comm.h` and add block comments for better context, will that be ok?

Reply:
After more thought, the naming stuff is previous baggage so let's save that for the end. I'll continue the review ignoring that for now.